openapi3/schema.go

```diff
@@ -1915,15 +1930,20 @@ func (schema *Schema) visitJSONObject(settings *schemaValidationSettings, value
 	return nil
 }
 
-func (schema *Schema) expectedType(settings *schemaValidationSettings, typ string) error {
+func (schema *Schema) expectedType(settings *schemaValidationSettings, value interface{}) error {
 	if settings.failfast {
 		return errSchema
 	}
+
+	a := "a"
+	if strings.Trim(schema.Type[:1], "aeiou") == "" {
```

Comment:
Please just compare to the few possible values explicitly with a switch statement.